### DISCUSSION TITLE
when use the CURLOPT_COPYPOSTFIELDS, how to release the copyed data?

### DISCUSSION THREAD

**Discussion by xinsuinizhuan** [curl/curl#9557] [Q&A] [answered]
i use the :
 curl_easy_setopt(easy_handle, CURLOPT_POST, 1L);
 curl_easy_setopt(easy_handle, CURLOPT_POSTFIELDSIZE, contentsize);
 curl_easy_setopt(easy_handle, CURLOPT_COPYPOSTFIELDS, content);
 curl_multi_add_handle(g_async_multi, easy_handle);

when call the curl_multi_perform, then how to release the copyed data?

**Comment by bagder** [comment, marked as answer]
All options set with `curl_easy_setopt` remain set until one of these actions are made:

- the easy handle is killed (using `curl_easy_cleanup()`
- the option is set again (to something else)
- `curl_easy_reset()` resets them back to default values

This includes all options that allocate and copy the data, such as `CURLOPT_COPYPOSTFIELDS`

**Reply by xinsuinizhuan** [reply]
OK. i see, call the curl_easy_cleanup() or curl_easy_reset() will release the copyed data.
